### ISSUE TITLE
Can't run pm2 as another user

### ISSUE THREAD

**Issue by Sir-Will** [Unitech/pm2#2033]
hey,

I'm trying to run pm2 as another user with this command:
`sudo -u scripts bash -c 'pm2 restart 2'`
but this still runs it as the logged in user.

**Comment by soyuka** [comment]
As long as the user you want to run pm2 from has the `pm2` binary in it's path it should work. Note that every pm2 instance has it's own memory daemon. This means that you should first start the pm2 instance from that user so that the in-memory process user is the correct one.

Take this screenshot for example:
![screenshot from 2016-03-23 11 57 18](https://cloud.githubusercontent.com/assets/1321971/13983160/5a8cf066-f0ee-11e5-8f4b-443983f8775a.png)

Note that I'm sourcing my `.zshrc` file because `www-data` has a `/usr/sbin/nologin` shell.

What I did for this to work:
1. Install nodejs through nvm from the `www-data` user. It now has it's own nodejs binary, with it's own nodejs version
2. Install pm2 from the `www-data` user, you can see on the screenshot that both users (me and `www-data`) are using different pm2 versions!
3. Be careful when you issue start commands, if a server is running from `www-data` you won't be able to start it from another user without changing it's listening port. It can also be disturbing because the `pm2 list` will only show processes managed by the user issuing the command (better for security).

**Comment by Sir-Will** [comment]
Hm, so I have to install nodejs and pm2 in the user who owns the scripts and in the user which will run the script as the other user?

**Comment by soyuka** [comment]
This is prefered to avoid errors. If the user has correct ownerships on ~/.pm2 of the other user you could have only one installation. I haven't tested this but it's not officially supported.

**Comment by Sir-Will** [comment]
I can run `pm2 show` fine on the "rundeck" user but I still can't run `sudo -u scripts bash -c 'pm2 show'`

I get this error if I run it:
```
Error: connect EACCES /home/rundeck/.pm2/rpc.sock
    at Object.exports._errnoException (util.js:870:11)
    at exports._exceptionWithHostPort (util.js:893:20)
    at PipeConnectWrap.afterConnect [as oncomplete] (net.js:1057:14)
```

**Comment by soyuka** [comment]
Add the `--hp` to specify the user home path or it'll take the current user `$HOME`.

**Comment by Sir-Will** [comment]
I'm not aware of a `--hp` option for sudo.
How do you mean it?

**Comment by rahulka** [comment]
Hi 😄 

The problem I am facing today also is similar (or identical)

```
jenkins@ip-172-xx-xx-xxx:~$ pm2 reload all -u ubuntu --hp /home/ubuntu
Use --update-env to update environment variables
[PM2][WARN] No process found

jenkins@ip-172-xx-xx-xxx:~$ exit
ubuntu@ip-172-xx-xx-xxx:~$ pm2 reload all
Use --update-env to update environment variables
[PM2] Applying action reloadProcessId on app [all](ids: 0,1,2,3)
[PM2] [app](0) ✓
[PM2] [app](1) ✓
[PM2] [app](3) ✓
[PM2] [app](2) ✓
┌──────────┬────┬─────────┬──────┬────────┬─────────┬────────┬──────┬───────────┬──────────┐
│ App name │ id │ mode    │ pid  │ status │ restart │ uptime │ cpu  │ mem       │ watching │
├──────────┼────┼─────────┼──────┼────────┼─────────┼────────┼──────┼───────────┼──────────┤
│ app      │ 0  │ cluster │ 7157 │ online │ 7       │ 1s     │ 55%  │ 89.6 MB   │ disabled │
│ app      │ 1  │ cluster │ 7158 │ online │ 7       │ 1s     │ 55%  │ 89.4 MB   │ disabled │
│ app      │ 2  │ cluster │ 7181 │ online │ 7       │ 1s     │ 104% │ 85.8 MB   │ disabled │
│ app      │ 3  │ cluster │ 7187 │ online │ 7       │ 0s     │ 104% │ 88.5 MB   │ disabled │
└──────────┴────┴─────────┴──────┴────────┴─────────┴────────┴──────┴───────────┴──────────┘
 Use `pm2 show <id|name>` to get more details about an app
```
Can you please provide an approach so that `jenkins` user (not a sudo-er, but can gain root access via a script by visudo configuration) can perform actions on pm2 running in other user's context.

Thanks in advance 😄

**Comment by Unitech** [comment]
Feature added in upcoming PM2. To try it:

(Make sure pm2 is run as root to manipulate uid and gid)

```bash
$ pm2 save
$ pm2 kill
$ npm install Unitech/pm2#development -g
$ sudo pm2 resurrect
$ sudo pm2 start app --uid <uid>
# Now pm2 ls display the user running the script
$ pm2 ls
```

Feedback is welcome

**Comment by Unitech** [comment]
Published on PM2 2.6.0:

```bash
$ npm install pm2@latest -g
$ pm2 update
```
